adapters/repos/db/repo.go

```diff
@@ -57,3 +57,9 @@ func (d *DB) GetIndex(kind kind.Kind, className schema.ClassName) *Index {
 
 	return index
 }
+
+// DeleteIndex deletes the index
```

Comment:
For my understanding: Is it correct that `.drop()` performs all the teardown operations and `.Delete()` only deletes the already cleaned up index? Is there a reason to do this in two independent steps? Would someone ever call only `.drop()` or only `.Delete()` without calling the other? 

I would have imagined that the public `Delete()` method is the "entrypoint" ,so to speak, which first performs the teardown procedures and finally deletes everything. So essentially `drop()` would be called from within `DeleteIndex()`, so an "outsider" such as `migrater.DropClass()` does not need to know that a `.drop()` method exists. Is there a specific reason not to do this?

Reply:
That's a good point, I'll move that drop code to `.Delete()`